### PR TITLE
Add SOT23 footprint and initial test

### PR DESCRIPTION
**Pull request by abhijitxy**
Add SOT23 package definition

### CODE REVIEW

**Comment by seveibar** [comment]
can you post a screenshot of what the footprint looks like? Note that we won't need to do this in the future if we integrate circuit-to-svg CC @imrishabh18

**Comment by seveibar** [comment]
I think the debug server is broken so we'll probably need to do that ASAP @imrishabh18, I can do it today if you're busy

**Comment by seveibar** [comment]
sorry @abhijitxy you're finding all the issues... our debug server literally just broke like 3 days ago so `logSoup` doesn't show you the visualization anymore T-T

**Comment by seveibar** [comment]
@imrishabh18 whenever somebody does `logSoup` we just need to run `circuit-to-svg` instead and have it drop an `svg`. We can just do it for a single test for now, so basically:

* introduce a `snapshotSoup(..)` function (ideally into a test fixture, i.e. `getTestFixture().snapshotSoup`
* Demonstrate usage with a single test

**Comment by imrishabh18** [comment]
Ohh okay, I was working on this issue https://github.com/tscircuit/footprinter/issues/14 and had made progress. I will push my update

**Comment by abhijitxy** [comment]
@seveibar please review it!!

File: src/fn/sot23.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    return { x: 1.7, y: -0.95 };
+  } else {
+    return { x: 1.7, y: 0.95 };
+  }
+};
+
+export const sot23WithoutParsing = (params: z.infer<typeof sot23_def>) => {
+  const pads: AnySoupElement[] = [];
+
+  for (let i = 0; i < params.num_pins; i++) {
```
Comment (seveibar) [inline src/fn/sot23.ts:38]: I _think_ sot23 always has 3 pins (maybe i'm wrong)

```suggestion
  for (let i = 0; i < 3; i++) {
```

**Review by seveibar — APPROVED** [review]
looks good, don't worry too much about the comment we can always adjust

**Pull request merged**